lib/cli/commands/generate.js

```diff
@@ -90,7 +52,12 @@ addons).
     let blueprintName = hasNamespace ? split.slice(1).join(':') : split[0];
     let addonData = discoverAddons(process.cwd());
     let addons = topsort(addonData).map((data) => {
-      return new Addon(data.name, data.dir);
+      let addonPath = path.join(data.dir, 'app', 'addon.js');
+      let Addon = require(addonPath).default;
```

Comment:
We might want to add a helpful error message if this fails to load, a la https://github.com/denali-js/denali/blob/master/lib/runtime/addon.js#L200